src/common/Util.js

```diff
@@ -1,11 +1,12 @@
 const debug = require('debug')('prismarine-auth')
 
-function checkStatus (res) {
+async function checkStatus (res) {
   if (res.ok) { // res.status >= 200 && res.status < 300
     return res.json()
   } else {
-    debug('Request fail', res)
-    throw Error(res.statusText)
+    const resp = await res.json()
+    debug('Request fail', resp)
```

Comment:
This may not always be a json. You might want to also add support for when it's just text, unless you've tested that.

Reply:
It should fail as expected in that case. None of the APIs here if operating correctly should return anything but json.